### PR TITLE
d_a_item -> Layout and Enum

### PR DESCRIPTION
**Pull request by elijah-thomas774**

### CODE REVIEW

File: include/d/a/d_a_itembase.h [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x00F */ ITEM_SAILCLOTH = 15,
+    /* 0x010 */ ITEM_GODDESS_HARP = 16,
+    /* 0x011 */ ITEM_SPIRIT_VESSEL = 17,
+    /* 0x012 */ ITEM_UNK_18 = 18,
+    /* 0x013 */ ITEM_BOW = 19,
+    /* 0x014 */ ITEM_CLAWSHOTS = 20,
+    /* 0x015 */ ITEM_BIRD_STATUETTE = 21,
+    /* 0x016 */ ITEM_DEKU_HORNET = 22,
+    /* 0x017 */ ITEM_UNK_23 = 23,
+    /* 0x018 */ ITEM_UNK_24 = 24,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:31]:
```suggestion
    /* 0x016 */ ITEM_COMMON_BUG = 22,
    /* 0x017 */ ITEM_UNCOMMON_BUG = 23,
    /* 0x018 */ ITEM_RARE_BUG = 24,
```

File: include/d/a/d_a_itembase.h [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x037 */ ITEM_MUSHROOM_SPORES = 55,
+    /* 0x038 */ ITEM_DIGGING_MITTS = 56,
+    /* 0x039 */ ITEM_5_DEKU_SEEDS = 57,
+    /* 0x03A */ ITEM_UNK_58 = 58,
+    /* 0x03B */ ITEM_UNK_59 = 59,
+    /* 0x03C */ ITEM_10_DEKU_SEEDS = 60,
+    /* 0x03D */ ITEM_DUSK_RELIC_MAYBE = 61,
+    /* 0x03E */ ITEM_DUSK_RELIC = 62,
+    /* 0x03F */ ITEM_EVIL_CRYSTAL = 63,
+    /* 0x040 */ ITEM_BLUE_FEATHER_GOLD_SKULL_PLUME = 64,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:71]:
```suggestion
    /* 0x03D */ ITEM_COMMON_TREASURE = 61,
    /* 0x03E */ ITEM_COMMON_TREASURE2 = 62,
    /* 0x03F */ ITEM_UNCOMMON_TREASURE = 63,
    /* 0x040 */ ITEM_RARE_TREASURE = 64,
```

File: include/d/a/d_a_itembase.h [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x041 */ ITEM_GUARDIAN_POTION = 65,
+    /* 0x042 */ ITEM_GUARDIAN_POTION_PLUS = 66,
+    /* 0x043 */ ITEM_UNK_67 = 67,
+    /* 0x044 */ ITEM_WATER_SCALE = 68,
+    /* 0x045 */ ITEM_UNK_69 = 69,
+    /* 0x046 */ ITEM_FAIRY = 70,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:77]:
```suggestion
    /* 0x046 */ ITEM_BUG_MEDAL = 70,
```

File: include/d/a/d_a_itembase.h [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x047 */ ITEM_BUG_NET = 71,
+    /* 0x048 */ ITEM_FAIRY_BUG_NET = 72,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:79]:
```suggestion
    /* 0x048 */ ITEM_FAIRY = 72,
```

File: include/d/a/d_a_itembase.h [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x04F */ ITEM_HEART_POTION_PLUS = 79,
+    /* 0x050 */ ITEM_HEART_POTION_ALSO = 80,
+    /* 0x051 */ ITEM_HEART_POTION_PLUS_PLUS = 81,
+    /* 0x052 */ ITEM_UNK_82 = 82,
+    /* 0x053 */ ITEM_GUARDIAN_POTION_NO_TBOX = 83,
+    /* 0x054 */ ITEM_STAMINA_POTION = 84,
+    /* 0x055 */ ITEM_STAMINA_POTION_PLUS = 85,
+    /* 0x056 */ ITEM_AIR_POTION = 86,
+    /* 0x057 */ ITEM_AIR_POTION_PLUS = 87,
+    /* 0x058 */ ITEM_FAIRY_FIRST = 88,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:95]:
```suggestion
    /* 0x058 */ ITEM_BOTTLED_FAIRY = 88,
```

File: include/d/a/d_a_itembase.h [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x097 */ ITEM_SKY_STAG_BEETLE = 151,
+    /* 0x098 */ ITEM_STARRY_FIREFLY = 152,
+    /* 0x099 */ ITEM_BOTTLE = 153,
+    /* 0x09A */ ITEM_RUPEE_MEDAL_ALSO = 154,
+    /* 0x09B */ ITEM_HEART_MEDAL_ALSO = 155,
+    /* 0x09C */ ITEM_UNK_156 = 156,
+    /* 0x09D */ ITEM_UNK_157 = 157,
+    /* 0x09E */ ITEM_CAWLINS_LETTER = 158,
+    /* 0x09F */ ITEM_HORNED_COLOSSOL_BEETLE = 159,
+    /* 0x0A0 */ ITEM_BABY_RATTLE = 160,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:167]:
```suggestion
    /* 0x09E */ ITEM_CAWLIN_LETTER = 158,
    /* 0x09F */ ITEM_BEEDLE_INSECT_CAGE = 159,
    /* 0x0A0 */ ITEM_RATTLE = 160,
```

This one's just to match the in-game names :p

File: include/d/a/d_a_itembase.h [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x0A1 */ ITEM_HORNET_LARVAE = 161,
+    /* 0x0A2 */ ITEM_BIRD_FEATHER = 162,
+    /* 0x0A3 */ ITEM_TUMBLE_WEED = 163,
+    /* 0x0A4 */ ITEM_LIZARD_TAIL = 164,
+    /* 0x0A5 */ ITEM_ELDIN_ORE = 165,
+    /* 0x0A6 */ ITEM_ANCIENT_FLOWER = 166,
+    /* 0x0A7 */ ITEM_AMBER_RELIC = 167,
+    /* 0x0A8 */ ITEM_DUSK_RELIC_AGAIN = 168,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:175]:
```suggestion
    /* 0x0A8 */ ITEM_DUSK_RELIC = 168,
```

File: include/d/a/d_a_itembase.h [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x0A9 */ ITEM_JELLY_BLOB = 169,
+    /* 0x0AA */ ITEM_MONSTER_CLAW = 170,
+    /* 0x0AB */ ITEM_MONSTER_HORN = 171,
+    /* 0x0AC */ ITEM_ORNAMENTAL_SKULL = 172,
+    /* 0x0AD */ ITEM_EVIL_CRYSTAL_ITEM = 173,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:180]:
```suggestion
    /* 0x0AD */ ITEM_EVIL_CRYSTAL = 173,
```

File: include/d/a/d_a_itembase.h [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+    /* 0x0C1 */ ITEM_SOTH = 193,
+    /* 0x0C2 */ ITEM_REVITALIZING_POTION_PLUS_PLUS = 194,
+    /* 0x0C3 */ ITEM_HOT_SOUP = 195,
+    /* 0x0C4 */ ITEM_COLD_SOUP = 196,
+    /* 0x0C5 */ ITEM_LIFE_TREE_SEED = 197,
+    /* 0x0C6 */ ITEM_LIFE_TREE_FRUIT = 198,
+    /* 0x0C7 */ ITEM_EXTRA_WALLET = 199,
+    /* 0x0C8 */ ITEM_UNK_200 = 200,
+    /* 0x18D */ ITEM_LIFE_MEDAL_AGAIN = 397,
+    /* 0x18E */ ITEM_POUCH_EXPANSION_GODDESS_CHEST = 398,
```
Comment (CovenEsme) [inline include/d/a/d_a_itembase.h:209]: Are these ones actually defined? Seems weird that they'd have 0 -> 200, 397, 398, 511

File: src/toBeSorted/special_item_drop_mgr.cpp [hunk 10]
```diff
@@ -366,34 +367,13 @@ const LowHealthReplacement LOW_HEALTH_REPLACEMENTS[] = {
...
-    72,   // Fairy
-    1,    // Small Key
-    34,   // Rupoor
-};
+int SPECIAL_ITEM_ARRAY[] = {ITEM_NONE,       ITEM_HEART,         ITEM_HEART,         ITEM_GREEN_RUPEE,
+                            ITEM_BLUE_RUPEE, ITEM_RED_RUPEE,     ITEM_FARORE_TEAR,   ITEM_5_BOMBS,
+                            ITEM_10_BOMBS,   ITEM_SINGLE_ARROW,  ITEM_BUNDLE_ARROWS, ITEM_5_DEKU_SEEDS,
+                            ITEM_NONE,       ITEM_NONE,          ITEM_KEY_PIECE,     ITEM_ELDIN_ORE,
+                            ITEM_UNK_58,     ITEM_UNK_59,        ITEM_UNK_183,       ITEM_UNK_184,
+                            ITEM_UNK_185,    ITEM_FAIRY_BUG_NET, ITEM_FAIRY_BUG_NET, ITEM_SMALL_KEY,
```
Comment (CovenEsme) [inline src/toBeSorted/special_item_drop_mgr.cpp:375]:
```suggestion
                            ITEM_UNK_185,    ITEM_FAIRY, ITEM_FAIRY, ITEM_SMALL_KEY,
```

Cos the item name changed in the enum

**Review by CovenEsme — COMMENTED** [review]
Updated some item names from HDR :p

File: include/d/a/d_a_itembase.h (hunk 9 above)
Comment (elijah-thomas774) [reply include/d/a/d_a_itembase.h:209]: I just copied what was from ghidra, i figured it couldn't hurt to have here. Ill put a comment in

**Comment by elijah-thomas774** [comment]
I dont think i currently have the motivation to tackle this one xD

File: src/REL/d/t/d_t_tumble_weed.cpp
Comment (robojumper) [inline src/REL/d/t/d_t_tumble_weed.cpp:1]: Something seems to have gone wrong with this file during a rebase

**Review by robojumper — CHANGES_REQUESTED** [review]

**Review by robojumper — COMMENTED** [review]
Aww, I was hoping we'd see `dAcItem_c::init` matched 😄

**Review by robojumper — APPROVED** [review]
wrong button

**Pull request merged**